sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/core/TemporaryTable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.pipeline.sql.app.core;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.impl.AbstractTable;
+
+/**
+ * Represents a Calcite temporary table created from a {@link org.apache.calcite.rel.RelNode}'s type class.
```

Comment:
Can you add more documentation on why we need this table? Since I am wondering if Calcite has some AbstractTable implementation we directly use

Reply:
I believe that this table is purely necessary to represent a temporary table for a given RelNode since the temporary table already knows what row data type to return. All of the current implementations of Table in Calcite require a concrete table type, whereas this TemporaryTable is simply a wrapper. I'll add some additional documentation for it though.